django_select2/admin.py

```diff
@@ -0,0 +1,133 @@
+# -*- coding: utf-8 -*-
+"""
+Django-Select2 Tools for Django Admin.
+
+    Sample usage:
+        from django_select2.admin import Select2ModelAdmin
+        from django_select2.forms import Select2Widget
+
+
+        class MyModelAdmin(Select2ModelAdmin):
+            select2_fields = {
+                'my_choice_field': {
+                    'widget': Select2Widget,
+                },
+                'my_FK_field': {
+                    'widget_kwargs': {
+                        'search_fields': ['name__icontains'],
+                        'attrs': {
+                            'data-minimum-input-length': 2,
+                            'style': 'width:200px;',
+                        },
+                    },
+                },
+            }
+"""
+from django.conf import settings
+from django.contrib import admin
+from django.db.models import ForeignKey, ManyToManyField
+
+from .forms import (
+    ModelSelect2Mixin, ModelSelect2MultipleWidget, ModelSelect2Widget,
+    Select2Widget
+)
+
+
+class Select2ModelAdminMixin(object):
+    """
+    Base mixin that allows to define and set select2 widgets on model fields.
+
+    It relies on some Django admin.options.BaseModelAdmin attributes and needs
+    a single configuration parameter to activate select2 on specific fields.
+
+    ``select_2_fields`` is a dictionary of model field names mapped to a
+    configuration dictionary which might be empty, to rely on a fully automatic
+    configuration, or containing one or all of the optional settings:
+        - ``widget``: user defined widget for that field (e.g. Select2Widget),
+          when not provided, the widget is assigned automatically;
+        - ``widget_kwargs``: custom arguments for widget initialization
+          (e.g. {'search_fields': ['name__icontains']}).
+
+    """
+
+    select2_fields = {}
+
+    def get_widgets(self):
+        """Assign select2 widgets as defined via the select2_fields dict."""
```

Comment:
Since I accepted this correction, I had not commented on it. Maybe this way it looks like I am still in doubt about it. I prefer pushing all changes together but, if you say so, I will first push this smaller fixes first.